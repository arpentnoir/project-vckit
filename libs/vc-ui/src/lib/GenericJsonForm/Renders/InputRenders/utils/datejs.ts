/**
 * Origin From
 * https://github.com/eclipsesource/jsonforms/blob/6a6af7e7aca7e44c7ba745edb8611f0b7d7acd77/packages/material/src/util/datejs.tsx
 */
import dayjs from 'dayjs';
import customParsing from 'dayjs/plugin/customParseFormat';

// required for the custom save formats in the date, time and date-time pickers
dayjs.extend(customParsing);

export const createOnChangeHandler = (
  path: string,
  handleChange: (path: string, value: any) => void,
  saveFormat: string | undefined
) => (time: dayjs.Dayjs | null, textInputValue?: string) => {
  if (!time) {
    handleChange(path, undefined);
    return;
  }
  const result = dayjs(time).format(saveFormat);
  handleChange(path, result === 'Invalid Date' ? textInputValue : result);
};
