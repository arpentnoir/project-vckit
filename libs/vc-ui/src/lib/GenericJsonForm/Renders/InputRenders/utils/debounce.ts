/**
 * Based on
 * https://github.com/eclipsesource/jsonforms/blob/6a6af7e7aca7e44c7ba745edb8611f0b7d7acd77/packages/material/src/util/debounce.ts
 */
import debounce from 'lodash/debounce';
import { useState, useCallback, useEffect } from 'react'


const eventToValue = (ev: React.ChangeEvent<HTMLInputElement>): string => ev.target.value;
export const useDebouncedChange = (handleChange: (path: string, value: any) => void, defaultValue: any, data: any, path: string, eventToValueFunction: (ev: any) => string = eventToValue, timeout = 300): [any, React.ChangeEventHandler, () => void] => {
  const [input, setInput] = useState(data ?? defaultValue);
  useEffect(() => {
    setInput(data ?? defaultValue);
  }, [data]);
  const debouncedUpdate = useCallback(debounce((newValue: string) => handleChange(path, newValue), timeout), [handleChange, path, timeout]);
  const onChange = useCallback((ev: React.ChangeEvent<HTMLInputElement>) => {
    const newValue = eventToValueFunction(ev);
    const setValue: string = newValue === '' || newValue.trim() === '' ? defaultValue : newValue;
    setInput(setValue);
    debouncedUpdate(setValue);
  }, [debouncedUpdate, eventToValueFunction]);
  const onClear = useCallback(() => { setInput(defaultValue); handleChange(path, undefined) }, [defaultValue, handleChange, path]);
  return [input, onChange, onClear];
};
