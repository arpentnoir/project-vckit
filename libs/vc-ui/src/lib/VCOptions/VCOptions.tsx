import {
  Box,
  FormControl,
  InputLabel,
  MenuItem,
  Select,
  SelectChangeEvent,
} from '@mui/material';
import { FunctionComponent, useEffect, useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { Button } from '../Button';
import { Text } from '../Text';

export interface Form {
  schema: Record<string, unknown>;
  uiSchema: Record<string, unknown>;
}

export interface FormOption {
  id: string;
  name: string;
  displayName: string;
  fullForm?: Form;
  partialForm?: Form;
  credentialType?: string;
  formType?: string;
}

interface VCOptionsProps {
  forms: FormOption[];
  onFormSelected: (value: FormOption) => void;
}

export const VCOptions: FunctionComponent<VCOptionsProps> = ({
  forms,
  onFormSelected,
}) => {
  const [availableForms, setAvailableForms] = useState<FormOption[]>([]);
  const [selectedDocument, setSelectedDocument] = useState('');
  const [selectedDocumentName, setSelectedDocumentName] = useState('');
  const [selectedForm, setSelectedForm] = useState('');
  const [selectedCredential, setSelectedCredential] = useState('');

  const navigate = useNavigate();

  const goToHome = () => {
    navigate('/');
  };

  useEffect(() => {
    setAvailableForms(forms);
  }, [forms]);

  const handleSelectedDocument = (event: SelectChangeEvent<string>) => {
    setSelectedDocument(event.target.value);
    setSelectedDocumentName(
      availableForms.filter((value) => value.id === event.target.value)[0].name
    );
  };

  const handleSubmit = () => {
    const form = availableForms.filter(
      (form) => form.id === selectedDocument
    )[0];

    const formWithTypes = {
      ...form,
      credentialType: selectedCredential,
      formType: selectedForm,
    };

    onFormSelected(formWithTypes);
  };

  const isComplete = selectedDocument && selectedForm && selectedCredential;

  return (
    <Box>
      <Text
        fontWeight="bold"
        variant="h5"
        sx={{
          paddingTop: '10px',
          paddingBottom: '15px',
        }}
      >
        Verifiable Credential Options
      </Text>
      <FormControl fullWidth sx={{ marginBottom: '20px' }}>
        <InputLabel
          id="form-select-label"
          htmlFor={'select-input-label-id'}
          aria-label={`Select a document to issue. ${
            selectedDocumentName ? `${selectedDocumentName} selected` : ''
          }`}
        >
          Document Type
        </InputLabel>

        <Select
          labelId="form-select-label"
          label={'Document Type'}
          value={selectedDocument}
          onChange={handleSelectedDocument}
          role={'listbox'}
          data-testid={'form-select-document-type'}
          inputProps={{ id: 'select-input-label-id' }}
        >
          {availableForms.map(({ id, displayName }, index: number) => (
            <MenuItem key={`menu-item-${index}`} value={id}>
              {displayName}
            </MenuItem>
          ))}
        </Select>
      </FormControl>

      <FormControl fullWidth sx={{ marginBottom: '20px' }}>
        <InputLabel
          id="form-select-label-form-type"
          htmlFor={'select-input-label-id-form-type'}
          aria-label={`Select a form type ${
            selectedForm === 'partial'
              ? 'Partial Form selected'
              : selectedForm === 'full'
              ? 'Full Form selected'
              : ''
          }`}
        >
          Form Type
        </InputLabel>

        <Select
          labelId="form-select-label-form-type"
          label={'Form Type'}
          value={selectedForm}
          onChange={(event) => setSelectedForm(event.target.value)}
          role={'listbox'}
          data-testid={'form-select-form-type'}
          inputProps={{ id: 'select-input-label-id-form-type' }}
        >
          <MenuItem key={`menu-item-form-full`} value={'full'}>
            Full Form
          </MenuItem>
          <MenuItem key={`menu-item-form-partial`} value={'partial'}>
            Partial Form
          </MenuItem>
        </Select>
      </FormControl>

      <FormControl fullWidth sx={{ marginBottom: '20px' }}>
        <InputLabel
          id="form-select-label-credential-type"
          htmlFor={'select-input-label-id-credential-type'}
          aria-label={`Select a credential type ${
            selectedCredential === 'oa'
              ? 'Open Attestation selected'
              : selectedCredential === 'svip'
              ? 'Silicon Valley Innovation Program selected'
              : ''
          }`}
        >
          Credential Type
        </InputLabel>

        <Select
          labelId="form-select-label-credential-type"
          label={'Credential Type'}
          value={selectedCredential}
          onChange={(event) => setSelectedCredential(event.target.value)}
          role={'listbox'}
          data-testid={'form-select-credential-type'}
          inputProps={{ id: 'select-input-label-id-credential-type' }}
        >
          <MenuItem key={`menu-item-form-oa`} value={'oa'}>
            Open Attestation
          </MenuItem>
          <MenuItem key={`menu-item-form-svip`} value={'svip'}>
            Silicon Valley Innovation Program
          </MenuItem>
        </Select>
      </FormControl>
      <Box
        display={'flex'}
        flexDirection={'row'}
        flexWrap={'wrap'}
        justifyContent="right"
        gap={'15px'}
      >
        <Button
          label="Cancel"
          variant="outlined"
          color="error"
          onClick={goToHome}
        />
        <Button
          label="Next"
          onClick={handleSubmit}
          disabled={isComplete ? false : true}
        />
      </Box>
    </Box>
  );
};
