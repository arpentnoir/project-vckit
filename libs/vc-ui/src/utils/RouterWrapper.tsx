import { FunctionComponent, Children, ReactNode } from 'react';
import { MemoryRouter, Routes } from 'react-router-dom';

interface RouterWrapperProps {
  children: ReactNode;
}
export const RouterWrapper: FunctionComponent<RouterWrapperProps> = ({
  children,
}) => {
  return (
    <MemoryRouter>
      <Routes>{Children.map(children, (child) => child)}</Routes>
    </MemoryRouter>
  );
};
