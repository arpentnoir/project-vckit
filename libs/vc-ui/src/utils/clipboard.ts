export const copyToClipboard = async (data: string) => {
  try {
    await navigator.clipboard.writeText(data);
  } catch (err) {
    throw new Error('Failed to copy data to the clipboard');
  }
};
