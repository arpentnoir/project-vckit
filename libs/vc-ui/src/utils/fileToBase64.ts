export interface FileData {
  dataURL: string;
  name: string;
  size: number;
  type: string;
}

export const fileToBase64 = async (
  file: File
): Promise<FileData | DOMException> => {
  const { name, size, type } = file;
  return new Promise((resolve, reject) => {
    const reader = new window.FileReader();
    reader.onerror = () => {
      reject(reader.error ?? '');
    };

    reader.onload = (event) => {
      resolve({
        dataURL: event?.target?.result,
        name,
        size,
        type,
      } as FileData);
    };
    reader.readAsDataURL(file);
  });
};
