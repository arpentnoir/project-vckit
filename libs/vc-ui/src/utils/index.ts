export * from './clipboard';
export * from './fileToBase64';
export * from './RouterWrapper';
export * from './test.utils';
export * from './vc';
