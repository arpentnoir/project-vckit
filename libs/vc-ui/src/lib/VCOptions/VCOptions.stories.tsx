import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { Route } from 'react-router-dom';
import { RouterWrapper } from '../../utils';
import { FormOption, VCOptions } from './VCOptions';

const Story: ComponentMeta<typeof VCOptions> = {
  component: VCOptions,
  title: 'VCOptions',
};
export default Story;

const Template: ComponentStory<typeof VCOptions> = (args) => (
  <RouterWrapper>
    <Route path="/*" element={<VCOptions {...args} />} />
  </RouterWrapper>
);

export const Primary = Template.bind({});

const form = {
  schema: {},
  uiSchema: {},
};

Primary.args = {
  forms: [{ id: '001', name: 'sampleForm', fullForm: form, partialForm: form }],
  onFormSelected: (value: FormOption) => {
    return;
  },
};
