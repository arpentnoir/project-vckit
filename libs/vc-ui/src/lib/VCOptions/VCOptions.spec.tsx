import { fireEvent, within } from '@testing-library/react';
import { render } from '../../utils';
import { VCOptions } from './VCOptions';

// eslint-disable-next-line @typescript-eslint/no-unsafe-return
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

const mockNavigate = jest.fn();
const handleSubmitMock = jest.fn();

const formMock = {
  schema: {},
  uiSchema: {},
};

const formsMock = [
  {
    id: '001',
    name: 'testForm',
    displayName: 'testForm',
    fullForm: formMock,
    partialForm: formMock,
  },
];

describe('VCOptions', () => {
  beforeEach(() => jest.clearAllMocks());

  it('should render correctly', () => {
    const { baseElement } = render(
      <VCOptions forms={formsMock} onFormSelected={handleSubmitMock} />
    );

    expect(baseElement).toMatchSnapshot();
  });

  describe('document type', () => {
    it('should display document types', () => {
      const { getAllByRole } = render(
        <VCOptions forms={formsMock} onFormSelected={handleSubmitMock} />
      );

      fireEvent.mouseDown(getAllByRole('listbox')[0].children[0]);
      const listbox = within(getAllByRole('listbox')[0]);

      expect(listbox.getByText(formsMock[0].name)).toBeTruthy();
    });

    it('should update the value of the listbox', () => {
      const { getAllByRole } = render(
        <VCOptions forms={formsMock} onFormSelected={handleSubmitMock} />
      );

      fireEvent.mouseDown(getAllByRole('listbox')[0].children[0]);
      const listbox = within(getAllByRole('listbox')[0]);
      fireEvent.click(listbox.getByText(formsMock[0].name));

      expect(getAllByRole('listbox')[0].querySelector('input')?.value).toBe(
        formsMock[0].id
      );
    });
  });

  describe('form type', () => {
    it('should display form types', () => {
      const { getAllByRole } = render(
        <VCOptions forms={formsMock} onFormSelected={handleSubmitMock} />
      );

      fireEvent.mouseDown(getAllByRole('listbox')[1].children[0]);
      const listbox = within(getAllByRole('listbox')[0]);

      expect(listbox.getByText('Partial Form')).toBeTruthy();
      expect(listbox.getByText('Full Form')).toBeTruthy();
    });

    it('should update the value of the listbox', () => {
      const { getAllByRole } = render(
        <VCOptions forms={formsMock} onFormSelected={handleSubmitMock} />
      );

      fireEvent.mouseDown(getAllByRole('listbox')[1].children[0]);
      const listbox = within(getAllByRole('listbox')[0]);
      fireEvent.click(listbox.getByText('Partial Form'));

      expect(getAllByRole('listbox')[1].querySelector('input')?.value).toBe(
        'partial'
      );

      fireEvent.click(listbox.getByText('Full Form'));

      expect(getAllByRole('listbox')[1].querySelector('input')?.value).toBe(
        'full'
      );
    });
  });

  describe('credential type', () => {
    it('should display credential types', () => {
      const { getAllByRole } = render(
        <VCOptions forms={formsMock} onFormSelected={handleSubmitMock} />
      );

      fireEvent.mouseDown(getAllByRole('listbox')[2].children[0]);
      const listbox = within(getAllByRole('listbox')[0]);

      expect(listbox.getByText('Open Attestation')).toBeTruthy();
      expect(
        listbox.getByText('Silicon Valley Innovation Program')
      ).toBeTruthy();
    });

    it('should update the value of the listbox', () => {
      const { getAllByRole } = render(
        <VCOptions forms={formsMock} onFormSelected={handleSubmitMock} />
      );

      fireEvent.mouseDown(getAllByRole('listbox')[2].children[0]);
      const listbox = within(getAllByRole('listbox')[0]);
      fireEvent.click(listbox.getByText('Open Attestation'));

      expect(getAllByRole('listbox')[2].querySelector('input')?.value).toBe(
        'oa'
      );

      fireEvent.click(listbox.getByText('Silicon Valley Innovation Program'));

      expect(getAllByRole('listbox')[2].querySelector('input')?.value).toBe(
        'svip'
      );
    });
  });

  it('should display both buttons', () => {
    const { getByText } = render(
      <VCOptions forms={formsMock} onFormSelected={handleSubmitMock} />
    );

    expect(getByText('Cancel')).toBeTruthy();
    expect(getByText('Next')).toBeTruthy();
  });

  it('should disable the next button if options are not selected', () => {
    const { getByText } = render(
      <VCOptions forms={formsMock} onFormSelected={handleSubmitMock} />
    );

    expect(getByText('Next').getAttribute('disabled'));
  });

  it('should call the callback function when the user clicks next', () => {
    const { getByText, getAllByRole } = render(
      <VCOptions forms={formsMock} onFormSelected={handleSubmitMock} />
    );

    fireEvent.mouseDown(getAllByRole('listbox')[0].children[0]);
    const documentListbox = within(getAllByRole('listbox')[0]);
    fireEvent.click(documentListbox.getByText(formsMock[0].name));

    fireEvent.mouseDown(getAllByRole('listbox')[1].children[0]);
    const formListBox = within(getAllByRole('listbox')[0]);
    fireEvent.click(formListBox.getByText('Full Form'));

    fireEvent.mouseDown(getAllByRole('listbox')[2].children[0]);
    const credentialListBox = within(getAllByRole('listbox')[0]);
    fireEvent.click(credentialListBox.getByText('Open Attestation'));

    fireEvent.click(getByText('Next'));

    expect(handleSubmitMock).toBeCalledWith({
      ...formsMock[0],
      formType: 'full',
      credentialType: 'oa',
    });
  });

  it('should go to home when cancel is clicked', () => {
    const { getByText } = render(
      <VCOptions forms={formsMock} onFormSelected={handleSubmitMock} />
    );

    fireEvent.click(getByText('Cancel'));

    expect(mockNavigate).toBeCalledWith('/');
  });
});
