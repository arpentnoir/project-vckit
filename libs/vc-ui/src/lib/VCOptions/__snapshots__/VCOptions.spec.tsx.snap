// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VCOptions should render correctly 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1sv2wdh-MuiTypography-root"
        tabindex="0"
      >
        Verifiable Credential Options
      </h5>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-1lof2qr-MuiFormControl-root"
      >
        <label
          aria-label="Select a document to issue. "
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="select-input-label-id"
          id="form-select-label"
        >
          Document Type
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          data-testid="form-select-document-type"
          role="listbox"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="form-select-label"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            role="button"
            tabindex="0"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            id="select-input-label-id"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Document Type
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-1lof2qr-MuiFormControl-root"
      >
        <label
          aria-label="Select a form type "
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="select-input-label-id-form-type"
          id="form-select-label-form-type"
        >
          Form Type
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          data-testid="form-select-form-type"
          role="listbox"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="form-select-label-form-type"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            role="button"
            tabindex="0"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            id="select-input-label-id-form-type"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Form Type
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-1lof2qr-MuiFormControl-root"
      >
        <label
          aria-label="Select a credential type "
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="select-input-label-id-credential-type"
          id="form-select-label-credential-type"
        >
          Credential Type
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          data-testid="form-select-credential-type"
          role="listbox"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="form-select-label-credential-type"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            role="button"
            tabindex="0"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            id="select-input-label-id-credential-type"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Credential Type
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiBox-root css-1oknf09"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1h2qilq-MuiButtonBase-root-MuiButton-root"
          data-testid="button:Cancel"
          tabindex="0"
          type="button"
        >
          <div
            class="css-doi3ij-MuiStack-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              data-testid="button:Cancel:label"
              tabindex="0"
            >
              Cancel
            </p>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1yt9wte-MuiButtonBase-root-MuiButton-root"
          data-testid="button:Next"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <div
            class="css-doi3ij-MuiStack-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              data-testid="button:Next:label"
              tabindex="0"
            >
              Next
            </p>
          </div>
        </button>
      </div>
    </div>
  </div>
</body>
`;
